ext/standard/array.c

```diff
@@ -5894,11 +5903,24 @@ PHPAPI bool php_array_pick_keys(const php_random_algo *algo, php_random_status *
 	zend_bitset_clear(bitset, bitset_len);
 
 	i = num_req;
+	int failures = 0;
 	while (i) {
 		randval = algo->range(status, 0, num_avail - 1);
-		if (!zend_bitset_in(bitset, randval)) {
+		if (EG(exception)) {
```

Comment:
I typically surround this with UNEXPECTED so that the compiler generates a nicer instruction ordering and place failure paths off of the hot path.

Reply:
This makes sense, but it would need to be consistently applied in the entirety of ext/random. I'll do this as a follow-up for master-only.